server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1558,6 +1577,11 @@ private static class IndexMetadataDiff implements Diff<IndexMetadata> {
                 DiffableUtils.getStringKeySerializer(),
                 ROLLOVER_INFO_DIFF_VALUE_READER
             );
+            if (in.getTransportVersion().onOrAfter(TransportVersions.INDEX_METADATA_MAPPINGS_UPDATED_VERSION)) {
+                mappingsUpdatedVersion = IndexVersion.readVersion(in);
+            } else {
+                mappingsUpdatedVersion = IndexVersions.ZERO;
```

Comment:
Should we be less optimistic and assume if we don't know the last version it was updated on, it is the current version?

Reply:
But it can't be the current version, else it would have that field in it? Maybe if the master is the previous version, but then the cluster wouldn't be using any new features (and the allocation running on master would ignore this anyway). Either way, defaulting to `current()` is _probably_ not correct.

Reply:
Could we use the lookup tables? Transport version -> Release version -> max(Index version)

Reply:
There's currently no way to get from release version to index version (as release version has no meaning)...